File: java/java-practise/src/main/java/day/five/selenium-page-object/src/test/java/pages/SubmittedPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class SubmittedPage {

    private final WebDriver driver;
    private final WebDriverWait wait;

    private final By messageText = By.id("message");

    public SubmittedPage(WebDriver driver, int waitInSeconds){
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(waitInSeconds));
        wait.until(ExpectedConditions.visibilityOfElementLocated(messageText));
    }

    public String getMessage(){
        return driver.findElement(messageText).getText();
    }

}

File: scharp/day/five/SeleniumTestProject/pages/SubmittedPage.cs
using OpenQA.Selenium;

namespace SeleniumTestProject.pages
{
    public class SubmittedPage : BasePage
    {
        private readonly By _messageText = By.Id("message");
        public SubmittedPage(IWebDriver driver, int waitTimeInSeconds): base(driver, waitTimeInSeconds) {
            wait.Until(driver =>
            {
                IWebElement messageText = driver.FindElement(_messageText);
                return messageText.Displayed ? messageText : null;
            });
        }

        public String MessageText()
        {
           return driver.FindElement(_messageText).Text;
        }
    }
}
